integration-cli: do not check creation datetime, it can change

Snorch opened the pull request:
In TestImagesFilterWhiteSpaceTrimmingAndLowerCasingWorking
if the time had changed between image listings, listings will be
different because they include creation time. Patch removes
creation time from comparison.
- check lenth != 0, in case of empty line

We got it running tests on v1.5.0 with vfs graph driver, for img
8ba30c03fa38 CREATED had changed from "3 minutes" to "4 minutes",
fail log:

```
out 0
<none>              <none>              77415989777c        4 minutes
ago       2.43 MB
<none>              <none>              8ba30c03fa38        3 minutes
ago       2.43 MB
<none>              <none>              d589e4d704ca        3 minutes
ago       2.43 MB
REPOSITORY          TAG                 IMAGE ID            CREATED
VIRTUAL SIZE
out 1
<none>              <none>              77415989777c        4 minutes
ago       2.43 MB
<none>              <none>              8ba30c03fa38        3 minutes
ago       2.43 MB
<none>              <none>              d589e4d704ca        3 minutes
ago       2.43 MB
REPOSITORY          TAG                 IMAGE ID            CREATED
VIRTUAL SIZE
out 2
<none>              <none>              77415989777c        4 minutes
ago       2.43 MB
<none>              <none>              8ba30c03fa38        4 minutes
ago       2.43 MB
<none>              <none>              d589e4d704ca        3 minutes
ago       2.43 MB
REPOSITORY          TAG                 IMAGE ID            CREATED
VIRTUAL SIZE
out 3
<none>              <none>              77415989777c        4 minutes
ago       2.43 MB
<none>              <none>              8ba30c03fa38        4 minutes
ago       2.43 MB
<none>              <none>              d589e4d704ca        3 minutes
ago       2.43 MB
REPOSITORY          TAG                 IMAGE ID            CREATED
VIRTUAL SIZE
out 4
<none>              <none>              77415989777c        4 minutes
ago       2.43 MB
<none>              <none>              8ba30c03fa38        4 minutes
ago       2.43 MB
<none>              <none>              d589e4d704ca        3 minutes
ago       2.43 MB
REPOSITORY          TAG                 IMAGE ID            CREATED
VIRTUAL SIZE
```

Signed-off-by: Pavel Tikhomirov ptikhomirov@parallels.com

Snorch committed:
integration-cli: do not check creation datetime, it can change

In TestImagesFilterWhiteSpaceTrimmingAndLowerCasingWorking
if the time had changed between image listings, listings will be
different because they include creation time. Patch removes
creation time from comparison.

- check lenth != 0, in case of empty line

We got it running tests on v1.5.0 with vfs graph driver, for img
8ba30c03fa38 CREATED had changed from "3 minutes" to "4 minutes",
fail log:
```
out 0
<none>              <none>              77415989777c        4 minutes
ago       2.43 MB
<none>              <none>              8ba30c03fa38        3 minutes
ago       2.43 MB
<none>              <none>              d589e4d704ca        3 minutes
ago       2.43 MB
REPOSITORY          TAG                 IMAGE ID            CREATED
VIRTUAL SIZE
out 1
<none>              <none>              77415989777c        4 minutes
ago       2.43 MB
<none>              <none>              8ba30c03fa38        3 minutes
ago       2.43 MB
<none>              <none>              d589e4d704ca        3 minutes
ago       2.43 MB
REPOSITORY          TAG                 IMAGE ID            CREATED
VIRTUAL SIZE
out 2
<none>              <none>              77415989777c        4 minutes
ago       2.43 MB
<none>              <none>              8ba30c03fa38        4 minutes
ago       2.43 MB
<none>              <none>              d589e4d704ca        3 minutes
ago       2.43 MB
REPOSITORY          TAG                 IMAGE ID            CREATED
VIRTUAL SIZE
out 3
<none>              <none>              77415989777c        4 minutes
ago       2.43 MB
<none>              <none>              8ba30c03fa38        4 minutes
ago       2.43 MB
<none>              <none>              d589e4d704ca        3 minutes
ago       2.43 MB
REPOSITORY          TAG                 IMAGE ID            CREATED
VIRTUAL SIZE
out 4
<none>              <none>              77415989777c        4 minutes
ago       2.43 MB
<none>              <none>              8ba30c03fa38        4 minutes
ago       2.43 MB
<none>              <none>              d589e4d704ca        3 minutes
ago       2.43 MB
REPOSITORY          TAG                 IMAGE ID            CREATED
VIRTUAL SIZE
```

Signed-off-by: Pavel Tikhomirov <ptikhomirov@parallels.com>

GordonTheTurtle commented:
Can you please sign your commits following these rules:

https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work

The easiest way to do this is to amend the last commit:

~~~console
$ git clone -b "do-not-check-creation-datetime" git@github.com:Snorch/docker.git somewhere
$ cd somewhere
$ git commit --amend -s --no-edit
$ git push -f
~~~
This will update the existing PR, so you do not need to open a new one.

Snorch commented:
Sorry, missed that.

Snorch committed:
debug: show intervals

It is temporary patch to see intervalls if the error occures in build
I can not reproduce such fail on my Fedora 20

Signed-off-by: Pavel Tikhomirov <ptikhomirov@parallels.com>

Snorch commented:
commit e39ea834a92154574be3dde67ae30acdb25b32f7 does almost the same thing, closing issue
